store/sqlstore/store.go

```diff
@@ -179,24 +181,62 @@ func New(settings model.SqlSettings, metrics einterfaces.MetricsInterface) *SqlS
 
 	store.initConnection()
 
-	if *settings.DriverName == model.DatabaseDriverPostgres {
-		ver, err := store.GetDbVersion(true)
-		if err != nil {
-			mlog.Critical("Cannot get DB version.", mlog.Err(err))
+	ver, err := store.GetDbVersion(true)
```

Comment:
Would it be feasible for the version check to become a separate function that either returns an error or simple boolean? It may look better overall and offer some simplifications (e.g. early returns, less logging calls).